### PR TITLE
Add SNMP agent plugin

### PR DESCRIPTION
**Pull request by MichalRebisz**
Snmp agent plugin manifest templates and tests.

### CODE REVIEW

File: manifests/plugin/snmp_agent.pp [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# https://wiki.opnfv.org/display/fastpath/SNMP+Agent+HLD
+class collectd::plugin::snmp_agent(
```
Comment (bastelfreak) [inline manifests/plugin/snmp_agent.pp]: can you please add puppet-strings documentation for the params?

File: templates/plugin/snmp_agent.conf.epp [hunk 2]
```diff
@@ -0,0 +1,107 @@
+<%- | Optional $data = undef,
```
Comment (bastelfreak) [inline templates/plugin/snmp_agent.conf.epp]: please add a comment like `# THIS FILE IS MANAGED BY PUPPET`

File: manifests/plugin/snmp_agent.pp (hunk 1 above)
Comment (MichalRebisz) [reply manifests/plugin/snmp_agent.pp]: Added

File: templates/plugin/snmp_agent.conf.epp (hunk 2 above)
Comment (MichalRebisz) [reply templates/plugin/snmp_agent.conf.epp]: Added

**Comment by bastelfreak** [comment]
@MichalRebisz can you please squash the commits down and ensure that the used email address in the commit is associated with your github account?

**Comment by MichalRebisz** [comment]
> @MichalRebisz can you please squash the commits down and ensure that the used email address in the commit is associated with your github account?

Commits squashed. Sorry about the account issue.

File: templates/plugin/snmp_agent.conf.epp (hunk 2 above)
Comment (bastelfreak) [reply templates/plugin/snmp_agent.conf.epp]: This template is broken since your squash. I think the newline needs to go away?

Comment (MichalRebisz) [reply templates/plugin/snmp_agent.conf.epp]: Your right sorry for that. Pushed a fix but the build failed on a totally different issue. It doesn't look like its caused by anything in the PR.

**Pull request merged**